gemini/sample-apps/fashion-trends-prediction/app/data_prep/driver.py

```diff
@@ -0,0 +1,331 @@
+import json
+import time
+
+from config import config
+from dataProcessing import getPosts
+from helper_functions_insta import getInfluencers, instalogin
+from helper_functions_vogue import get_articles
+from prepare_data_for_retriever import prepare_data_for_retriever
+
+parameters = config["parameters"]["standard"]
+country_names = config["countryList"]
+data_path = config["Data"]["current_data"]
+
+
+def isRoot(word):
+    if word.strip() == word and word.lower() == word:
+        return True
+    return False
+
+
+def cmpf(item):
+    return -len(item[1])
+
+
+def extract_json_influencer_wise(scrape_size, first_scrape, saved):
+    """Extracts data from Instagram and saves it in a JSON file.
+
+    Args:
+            scrape_size (dict): A dictionary containing the number of countries, influencers, and posts to scrape.
+            first_scrape (bool): A boolean indicating whether this is the first scrape.
+            saved (dict): A dictionary containing the saved data.
+
+    """
+
+    if first_scrape:
```

Comment:
Why do you need to specify a first scrape? Couldn't you provide a parameter for the existing data and check if it's empty?

Reply:
Removed this variable